### PR TITLE
#1306 add test that reproduces the flaky problem with CopyOnWriteMap

### PR DESCRIPTION
**Pull request by asolntsev**
it's a simplified version of what happens with `FakeValuesService.MAP_OF_METHOD_AND_COERCED_ARGS`.

@snuyanzin Can you help with fixing `CopyOnWriteMap`?

### CODE REVIEW

**Comment by bodiam** [comment]
Hi @asolntsev , how about getting rid of the COWMap? I just did a performance test with Spring Boot native, and I did about 300.000 requests, and in my "production" code I'm getting nullpointers. For some reason, I got more NPEs in GraalVM than while using a JVM. If you want, you can easily reproduce it here by running the k6 performance test: https://github.com/datafaker-net/datafaker-native-demo

**Comment by kingthorin** [comment]
Wasn't the COWMap originally introduced to address performance issues?

**Comment by bodiam** [comment]
@kingthorik I believe the idea is that it has some benefits regarding performance, but at the cost that the library no longer works reliably.

**Comment by asolntsev** [comment]
> how about getting rid of the COWMap? 

Yes, I like the idea. 
I even tried to do it in https://github.com/datafaker-net/datafaker/pull/1203

The argument against it was that COWMap is faster. But I never checked this statement by myself. And I don't think this speed is really important.

**Comment by kingthorin** [comment]
Will using a synchronized map alleviate the issue? I know that might cause a slight performance hit but hopefully it still facilitates the original case.

**Comment by asolntsev** [comment]
@kingthorin @bodiam I investigated the problem a bit more. 

Conclusion:
the problem is not in `CopyOnWriteMap`, but it's used. 

When we have map of maps, then we often call `get`, then `if != null` and `put`. This sequence doesn't work well in concurrent execution by multiple threads. Instead, we should use `computeIfAbsent`.

I submitted PR https://github.com/datafaker-net/datafaker/pull/1381 that should fix our flaky problem.

File: .run/tests.run.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <option name="MAIN_CLASS_NAME" value="" />
+    <option name="METHOD_NAME" value="" />
+    <option name="TEST_OBJECT" value="directory" />
+    <option name="VM_PARAMETERS" value="-ea -Duser.language=TR -Duser.country=tr" />
+    <dir value="$PROJECT_DIR$/src/test/java/net/datafaker" />
+    <method v="2">
+      <option name="Make" enabled="true" />
+    </method>
+  </configuration>
+</component>
```
Comment (bodiam) [inline .run/tests.run.xml:14]: do we need this file?

**Review by bodiam — APPROVED** [review]

**Comment by asolntsev** [comment]
The problem was fixed in PR https://github.com/datafaker-net/datafaker/pull/1381.
This PR is not needed anymore.

**Pull request closed without merging**